OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -489,16 +484,31 @@ Action InitHotkeysPanel(Widget panel)
 					var types = FieldLoader.GetValue<string[]>("Types", typesNode.Value.Value);
 					var added = new HashSet<HotkeyDefinition>();
 					var template = templates.Get(templateNode.Value.Value);
-					var remapDialogRoot = panel.Get("HOTKEY_DIALOG_ROOT");
-					var remapDialogPlaceholder = panel.GetOrNull("HOTKEY_DIALOG_PLACEHOLDER");
+
 					foreach (var t in types)
+					{
 						foreach (var hd in modData.Hotkeys.Definitions.Where(k => k.Types.Contains(t)))
+						{
 							if (added.Add(hd))
-								BindHotkeyPref(hd, modData.Hotkeys, template, hotkeyList, remapDialogRoot, remapDialogPlaceholder);
+							{
+								if (isFirstHotkey)
+								{
+									selectedHotkey = hd;
+									isFirstHotkey = false;
+								}
+
+								BindHotkeyPref(hd, modData.Hotkeys, template, hotkeyList, remapDialogRoot);
+							}
+						}
+					}
 				}
 			}
 
-			return () => { };
+			return () =>
+			{
+				selectedHotkey = null;
+				remapDialogRoot.RemoveChildren();
```

Comment:
Replace this with something like
```csharp
// Force the hotkey selection widget, if active, to yield focus
// HACK: we don't manage these widgets, so have to make a fragile guess
var w = Ui.KeyboardFocusWidget;
while (w != null)
{
	if (w == remapDialogRoot)
		Ui.KeyboardFocusWidget.YieldKeyboardFocus();

	w = w.Parent;
}
```
to work around the disappearing dialog issue.